src/App.vue

```diff
@@ -504,25 +503,35 @@ export default class App extends Mixins(DateMixin, FilingTemplateMixin, LegalApi
 
       // ensure NR is valid
       if (!this.isNrValid(nrResponse)) {
+        console.log('NR is not valid') // eslint-disable-line no-console
         this.nameRequestInvalidType = NameRequestStates.INVALID
         this.nameRequestInvalidErrorDialog = true
         return
       }
 
-      // TODO: verify that nrResponse.entityTypeCd === business.legalType
+      // ensure types match
+      if (this.nrTypeToEntityType(nrResponse) !== this.getEntityType) {
```

Comment:
Is this to protect against NR Type changes? 
Thats the only scenario i can think of where this might get out of sync, ie an IA is created under an original nr type which is then edited later?

Reply:
This was a left-over TODO from very early one. I just figured it was wise to verify that the correct NR type is being used.

Also I think we once said that the NR type would determine the business type, although I think that's now determined earlier in the process (ie, when affiliating the NR or creating the draft IA).

Your use case is a good example of a check we'd want. We'd also possibly that check in the MB table or entity dashboard.

Reply:
Interesting! Lets leave this comment open but don't let it hold your PR back 👍